addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/data/Thermostat.java

```diff
@@ -19,143 +25,159 @@
  */
 public class Thermostat extends BaseNestDevice {
     @SerializedName("can_cool")
-    private boolean canCool;
+    private Boolean canCool;
     @SerializedName("can_heat")
-    private boolean canHeat;
+    private Boolean canHeat;
     @SerializedName("is_using_emergency_heat")
-    private boolean isUsingEmergencyHeat;
+    private Boolean isUsingEmergencyHeat;
     @SerializedName("has_fan")
-    private boolean hasFan;
+    private Boolean hasFan;
     @SerializedName("fan_timer_active")
-    private boolean fanTimerActive;
+    private Boolean fanTimerActive;
     @SerializedName("fan_timer_timeout")
     private Date fanTimerTimeout;
     @SerializedName("has_leaf")
-    private boolean hasLeaf;
+    private Boolean hasLeaf;
     @SerializedName("temperature_scale")
-    private String tempScale;
+    private String temperatureScale;
     @SerializedName("ambient_temperature_c")
-    private Double ambientTemperature;
+    private Double ambientTemperatureC;
+    @SerializedName("ambient_temperature_f")
+    private Double ambientTemperatureF;
     @SerializedName("humidity")
     private Integer humidity;
     @SerializedName("target_temperature_c")
-    private Double targetTemperature;
+    private Double targetTemperatureC;
+    @SerializedName("target_temperature_f")
+    private Double targetTemperatureF;
     @SerializedName("target_temperature_high_c")
-    private Double targetTemperatureHigh;
+    private Double targetTemperatureHighC;
+    @SerializedName("target_temperature_high_f")
+    private Double targetTemperatureHighF;
     @SerializedName("target_temperature_low_c")
-    private Double targetTemperatureLow;
+    private Double targetTemperatureLowC;
+    @SerializedName("target_temperature_low_f")
+    private Double targetTemperatureLowF;
     @SerializedName("hvac_mode")
     private Mode mode;
     @SerializedName("previous_hvac_mode")
     private Mode previousMode;
     @SerializedName("hvac_state")
     private State state;
     @SerializedName("is_locked")
-    private boolean isLocked;
+    private Boolean isLocked;
     @SerializedName("locked_temp_max_c")
-    private Double lockedTemperatureHigh;
+    private Double lockedTemperatureHighC;
+    @SerializedName("locked_temp_max_f")
+    private Double lockedTemperatureHighF;
     @SerializedName("locked_temp_min_c")
-    private Double lockedTemperatureLow;
+    private Double lockedTemperatureLowC;
+    @SerializedName("locked_temp_min_f")
+    private Double lockedTemperatureLowF;
     @SerializedName("sunlight_correction_enabled")
-    private boolean sunlightCorrectionEnabled;
+    private Boolean sunlightCorrectionEnabled;
```

Comment:
Why?

Reply:
I've converted the properties in all these data objects from primitives to objects so it will be clear when data is missing (null value). Otherwise the binding will update the channel state to "false" when data is missing.